### PR TITLE
Improve error handling + record events

### PR DESCRIPTION
**Pull request by leochr**
**What this PR does / why we need it?**:

- Add multiple shortnames
- Improve error handling
- Record events
- Display failure reason and message
- Minor updates

### CODE REVIEW

File: deploy/crds/openliberty.io_openlibertyapplications_cr.yaml [hunk 1]
```diff
@@ -2,6 +2,7 @@ apiVersion: openliberty.io/v1beta1
 kind: OpenLibertyApplication
 metadata:
   name: demo-app
+  namespace: default
```
Comment (arturdzm) [inline deploy/crds/openliberty.io_openlibertyapplications_cr.yaml]: We probably should remove this from all example CRs to make it use current oc project

File: pkg/controller/openlibertytrace/openlibertytrace_controller.go [hunk 2]
```diff
@@ -233,16 +228,25 @@ func (r *ReconcileOpenLibertyTrace) UpdateStatus(conditionType openlibertyv1beta
 	// Keep the old `LastTransitionTime` when pod and status have not changed
```
Comment (arturdzm) [inline pkg/controller/openlibertytrace/openlibertytrace_controller.go]: Maybe we can re-use the openlibertyv1beta1.SetOperationCondtion method here

File: deploy/crds/openliberty.io_openlibertyapplications_cr.yaml (hunk 1 above)
Comment (leochr) [reply deploy/crds/openliberty.io_openlibertyapplications_cr.yaml]: Removed namespaces from all CRs.

File: pkg/controller/openlibertytrace/openlibertytrace_controller.go (hunk 2 above)
Comment (leochr) [reply pkg/controller/openlibertytrace/openlibertytrace_controller.go]: There is some additional logic in this method due to the fact that pod could change in CR.

**Pull request merged**